generic3g/OuterMetaComponent.F90

```diff
@@ -415,6 +428,14 @@ end subroutine set_entry_point
       module procedure recurse_
    end interface recurse
 
+   interface recurse_read_restart
+      module procedure recurse_read_restart_
+   end interface recurse_read_restart
+
```

Comment:
If you need these to be public then do they need to be type-bound as well?

Reply:
I think you may have duplicated superfluous bits of code in the existing source.   Please try deleting this and similar constructs and see if things still work.